command/version_history.go

```diff
@@ -0,0 +1,127 @@
+package command
+
+import (
+	"fmt"
+	"github.com/mitchellh/cli"
+	"github.com/posener/complete"
+	"github.com/ryanuber/columnize"
+	"strings"
+)
+
+var (
+	_ cli.Command             = (*VersionHistoryCommand)(nil)
+	_ cli.CommandAutocomplete = (*VersionHistoryCommand)(nil)
+)
+
+// VersionHistoryCommand is a Command implementation prints the version.
+type VersionHistoryCommand struct {
+	*BaseCommand
+}
+
+func (c *VersionHistoryCommand) Synopsis() string {
+	return "Prints the version history of the target Vault server"
+}
+
+func (c *VersionHistoryCommand) Help() string {
+	helpText := `
+Usage: vault version-history
+
+  Prints the version history of the target Vault server.
+
+  Print the version history:
+
+      $ vault version-history
+` + c.Flags().Help()
+	return strings.TrimSpace(helpText)
+}
+
+func (c *VersionHistoryCommand) Flags() *FlagSets {
+	return c.flagSet(FlagSetOutputFormat)
+}
+
+func (c *VersionHistoryCommand) AutocompleteArgs() complete.Predictor {
+	return complete.PredictNothing
+}
+
+func (c *VersionHistoryCommand) AutocompleteFlags() complete.Flags {
+	return c.Flags().Completions()
+}
+
+const versionTrackingWarning = "Note: Version tracking was added in 1.9.0. Earlier versions have not been tracked."
+
+func (c *VersionHistoryCommand) Run(args []string) int {
+	f := c.Flags()
+
+	if err := f.Parse(args); err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+
+	client, err := c.Client()
+	if err != nil {
+		c.UI.Error(err.Error())
+		return 2
+	}
+
+	resp, err := client.Logical().List("sys/version-history")
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Error reading version history: %s", err))
```

Comment:
Should we also mention that it was only added in 1.10 here?   We could also use the seal-status endpoint to detect the server version and not bother with the version-history call if the server is <1.10, but I'm not sure I want to set that precedent.

Reply:
I don't think that I like falling back to the seal-status endpoint either. I think mentioning that the version-history endpoint was added in 1.10 as well makes sense.

Reply:
I didn't add this warning directly to the error handling logic but instead to the existing warning at the top of the CLI output that mentions the earliest version that has been tracked.

See this commit https://github.com/hashicorp/vault/pull/13766/commits/7417fbb168f7bf1acccf19b8844accf2805eda85. What do you think?